presto-main/src/main/java/com/facebook/presto/util/MoreMath.java

```diff
@@ -79,6 +81,40 @@ public static double max(double... values)
                 .getAsDouble();
     }
 
+    /**
+     * Find minimum value from finite arguments, and return NaN otherwise.
+     */
+    public static double minExcludingNaNs(double... values)
+    {
+        double min = NaN;
+        for (double v : values) {
+            if (isFinite(v)) {
+                if (isNaN(min)) {
+                    min = v;
+                }
+                min = Math.min(min, v);
+            }
+        }
+        return min;
+    }
+
+    /**
+     * Find maximum value from finite arguments, and return NaN otherwise.
+     */
+    public static double maxExcludingNaNs(double... values)
```

Comment:
Returns the maximum value of the arguments. Returns NaN if there are no arguments or all arguments are NaN.